trollflow2/plugins/__init__.py

```diff
@@ -573,6 +585,29 @@ def _get_sunlight_coverage(area_def, start_time, overpass=None):
         return daylight_area / total_area
 
 
+def _get_product_area_def(job, area, product):
+    """Get area definition for a product."""
+    try:
+        if 'resampled_scenes' in job:
+            scn = job['resampled_scenes'][area]
+        else:
+            scn = job['scene']
+
+        if isinstance(product, tuple):
+            prod = scn[product[0]]
+        else:
+            prod = scn[product]
+    except KeyError:
+        try:
+            prod = scn[list(scn.keys())[0]]
```

Comment:
What happens if some of the datasets aren't complete (like 3a or 3b on avhrr ?)

Reply:
I guess then the user could move the plugin to be run after resampling.